packages/lib/src/components/internal/Address/components/AddressSearch.tsx

```diff
@@ -55,14 +71,32 @@ export default function AddressSearch({ onAddressLookup, onSelect, onManualAddre
         setErrorMessage(externalErrorMessage);
     }, [externalErrorMessage]);
 
-    const onChange = event => {
+    const onChange = async event => {
         if (!event.target.value) {
             setErrorMessage(i18n.get('address.errors.incomplete'));
             return;
         }
         const value = originalData.find(item => item.id === event.target.value);
-        onSelect(value);
-        setFormattedData([]);
+
+        // 1. in case we don't get a function just select item
+        if (typeof onAddressSelected !== 'function') {
+            onSelect(value);
+            setFormattedData([]);
+            return;
+        }
+
+        // 2. in case callback is provided, create and call onAddressSelected
+        new Promise<AddressLookupItem>((resolve, reject) => {
+            onAddressSelected(value, { resolve, reject });
+        })
+            .then(fullData => {
+                onSelect(fullData);
+                setFormattedData([]);
+                return;
+            })
+            .catch(reason => {
+                setErrorMessage(reason);
+            });
```

Comment:
Can we add a loading here?

Reply:
We have investigated this, yet we couldn't come up with a good solution. For example, Google own address solution doesn't have a loading indicator.
I will give the feedback to design and product.